File: src/renderer/light.zig
const clr = @import("../nmath/color.zig");
const vec = @import("../nmath/vector.zig");
const std = @import("std");
const expect = std.testing.expect;

pub const PointLight = struct {
    position: vec.Point,
    intensity: clr.Color,

    pub fn init(position: vec.Point, intensity: clr.Color) PointLight {
        return .{
            .position = position,
            .intensity = intensity,
        };
    }
};

File: pkg/renderer/light.go
package renderer

import "github.com/novelalex/soft-raytracer/pkg/nmath"

type PointLight struct {
	position  nmath.Vec3
	intensity nmath.Color
}

func NewPointLight(position nmath.Vec3, intensity nmath.Color) PointLight {
	return PointLight{position, intensity}
}
